op-node/rollup/derive/engine_controller.go

```diff
@@ -258,6 +291,22 @@ func (e *EngineController) TryUpdateEngine(ctx context.Context) error {
 }
 
 func (e *EngineController) InsertUnsafePayload(ctx context.Context, envelope *eth.ExecutionPayloadEnvelope, ref eth.L2BlockRef) error {
+	// Check if there is a finalized head once when doing EL sync. If so, transition to CL sync
+	if e.syncStatus == syncStatusWillStartEL {
+		b, err := e.engine.L2BlockRefByLabel(ctx, eth.Finalized)
+		if errors.Is(err, ethereum.NotFound) {
+			e.syncStatus = syncStatusStartedEL
+			e.log.Info("Starting EL sync")
+			e.elStart = time.Now()
```

Comment:
Ideally we'd use the clock abstraction here rather than time directly. It is just logging but nice to make progress towards abstracting time everywhere rather than away from it.

Reply:
I switched to using the clock abstraction internally (but haven't threaded it in yet)